### PR TITLE
Fixed sitemap generation for multiple frontend locales

### PR DESCRIPTION
**Pull request by sintro**
Fixes to sitemap/index.xml.builder for correct generation when multiple frontend locales are activated. 
More info in this https://github.com/refinery/refinerycms/issues/3215 issue.

### CODE REVIEW

File: core/app/views/refinery/sitemap/index.xml.builder [hunk 1]
```diff
@@ -8,10 +8,16 @@ xml.urlset "xmlns" => "http://www.sitemaps.org/schemas/sitemap/0.9" do
...
      elsif page.url.to_s !~ /^http/
        # handle relative link_url addresses.
-       [request.protocol, request.host_with_port, page.url].join
+       raw_url = [request.protocol, request.host_with_port, page.url].join
+       if (@locales.size > 1) && defined?(RoutingFilter::RefineryLocales)
+         filter = RoutingFilter::RefineryLocales.new
+         filter.around_generate({}) do
+           raw_url
+         end
+       end
```
Comment (parndt) [inline core/app/views/refinery/sitemap/index.xml.builder]: It doesn't seem like `raw_url` is being returned anymore if this condition fails, as it's not being returned in the `else` case.

**Comment by parndt** [comment]
Thanks for taking a look at this!

File: core/app/views/refinery/sitemap/index.xml.builder (hunk 1 above)
Comment (sintro) [inline core/app/views/refinery/sitemap/index.xml.builder]: You are right! Need to add else case to return raw_url without changes, or do some refactoring to get more effective correct code.

File: core/app/views/refinery/sitemap/index.xml.builder [hunk 2]
```diff
@@ -17,6 +17,8 @@ xml.urlset "xmlns" => "http://www.sitemaps.org/schemas/sitemap/0.9" do
          filter.around_generate({}) do
            raw_url
          end
+       else
+         raw_url
```
Comment (sintro) [inline core/app/views/refinery/sitemap/index.xml.builder]: Here is the fix, may be there are any better way to do it?

Comment (sintro) [inline core/app/views/refinery/sitemap/index.xml.builder]: So, here is the fix, may be there are any better way to do it?

**Comment by parndt** [comment]
Good enough for me - @bricesanchez ?

**Comment by bricesanchez** [comment]
Could you squash and rebase your branch ?

**Comment by parndt** [comment]
Done as a merge step

**Pull request merged**